tests/test_date_time_helpers.py

```diff
@@ -0,0 +1,183 @@
+import unittest
+
+from datetime import datetime as D
+from datetime import timedelta as T
+
+import numpy as np
+
+from exetera.processing import date_time_helpers as dth
+
+
+class S:
+    def __init__(self, start, end, expected=None, message=None):
+        self.start = start
+        self.end = end
+        self.expected = expected
+        self.message = message
+
+
+class TestDateTimeHelpers(unittest.TestCase):
+
+    def _do_scenario_test(self, scenarios, period, count):
+        for s in scenarios:
+            if isinstance(s.expected, (list, tuple)):
+                actual = dth.get_periods(s.start, s.end, period, count)
+                err = "{}, {}, {}, {}".format(s.start, s.end, period, count)
+                self.assertListEqual(s.expected, actual, err)
+            elif issubclass(type(s.expected), BaseException):
+                if s.message is not None:
+                    with self.assertRaises(s.expected, msg=s.message):
+                        dth.get_periods(s.start, s.end, period, count)
+                with self.assertRaises(s.expected):
+                    dth.get_periods(s.start, s.end, period, count)
+
+    def test_periods_zero_count(self):
+        scenarios = [
+            S(D(2020, 5, 10), D(2020, 5, 11), ValueError, "'delta' cannot be 0")
+        ]
+        self._do_scenario_test(scenarios, 'day', 0)
+
+
+    def test_periods_day_periods(self):
+
+        scenarios = [
+            S(D(2020, 5, 10), D(2020, 5, 9), ValueError),
+            S(D(2020, 5, 10), D(2020, 5, 10), [D(2020, 5, 10)]),
+            S(D(2020, 5, 10), D(2020, 5, 11), [D(2020, 5, 10), D(2020, 5, 11)])
+        ]
+        self._do_scenario_test(scenarios, 'day', 1)
+        self._do_scenario_test(scenarios, 'days', 1)
+
+        scenarios = [
```

Comment:
I would break up these test cases into separate cases testing positive and negative deltas. It's easier to identify what failed (positive or negative delta) when they're separate tests in the output and lets you run test individually and avoid testing something you don't need to check.